lib/logqueue-fifo.c

```diff
@@ -103,6 +103,19 @@ typedef struct _LogQueueFifo
  * log_queue_fifo_push_head() or log_queue_fifo_rewind_backlog().
  *
  */
+
+static void
+iv_list_update_msg_size(LogQueueFifo *self, struct iv_list_head *head)
```

Comment:
hmmm... I remember that we talked about this function IRL.
What do you think about storing the aggregated size per input queue(add a `gsize memory_usage` variable to the qoverflow_input embedded struct)? And when the input queue is full, just update the memory_usage counter in one step.